barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.cpp

```diff
@@ -255,12 +392,29 @@ void handle_arithmetic(Program::Opcode::AssertZero const& arg, AcirFormat& af, s
             af.original_opcode_indices.poly_triple_constraints.push_back(opcode_index);
         }
     } else {
-        af.quad_constraints.push_back(serialize_mul_quad_gate(arg.value));
-        af.original_opcode_indices.quad_constraints.push_back(opcode_index);
+        std::vector<mul_quad_<fr>> mul_quads;
+        // We try to use a single mul_quad gate to represent the expression.
+        if (arg.value.mul_terms.size() <= 1) {
+            auto quad = serialize_mul_quad_gate(arg.value);
+            // add it to the result vector if it worked
+            if (quad.a != 0 || !(quad.mul_scaling == fr(0)) || !(quad.a_scaling == fr(0))) {
+                mul_quads.push_back(quad);
+            }
+        }
+        if (mul_quads.empty()) {
+            // If not, we need to split the expression into multiple gates
+            mul_quads = serialize_big_add_gate(arg.value);
```

Comment:
Serialize_big_add_gate is a misnomer then, since you are serializing into a series of such gates